src/sentry/ratelimits/redis.py

```diff
@@ -17,30 +22,68 @@ def __init__(self, **options):
         cluster_key = getattr(settings, "SENTRY_RATE_LIMIT_REDIS_CLUSTER", "default")
         self.client = redis.redis_clusters.get(cluster_key)
 
+    def _construct_redis_key(
+        self, key: str, project: Project | None = None, window: int | None = None
+    ) -> str:
+        """
+        Construct a rate limit key using the args given. Key will have a format of:
+        "rl:<key_hex>:[project?<project_id>:]<time_bucket>"
+        where the time bucket is calculated by integer dividing the current time by the window
+        """
+
+        if window is None or window == 0:
+            window = self.window
+
+        key_hex = md5_text(key).hexdigest()
+        bucket = int(time() / window)
+
+        redis_key = f"rl:{key_hex}"
+        if project is not None:
+            redis_key += f":{project.id}"
+        redis_key += f":{bucket}"
+
+        return redis_key
+
     def validate(self):
         try:
             self.client.ping()
         except Exception as e:
             raise InvalidConfiguration(str(e))
 
-    def is_limited(self, key, limit, project=None, window=None):
-        if window is None:
-            window = self.window
+    def is_limited(self, key, limit, project=None, window=None) -> bool:
+        is_limited, _ = self.is_limited_with_value(key, limit, project=project, window=window)
+        return is_limited
 
-        key_hex = md5_text(key).hexdigest()
-        bucket = int(time() / window)
+    def current_value(self, key: int, project: Project = None, window: int = None) -> int:
+        """
+        Get the current value stored in redis for the rate limit with key "key" and said window
+        """
+        redis_key = self._construct_redis_key(key, project=project, window=window)
+
+        try:
+            current_count = self.client.get(redis_key)
+            if current_count is None:
+                # Key hasn't been created yet, therefore no hits done so far
+                return 0
+            return int(current_count)
+        except RedisError:
```

Comment:
We should only include the minimal set of lines that can throw in the `try` block:
```py
try:
    current_count = self.client.get(redis_key)
except RedisError:
    # Don't report any existing hits when there is a redis error.
    # Log what happened and move on
    logger.exception("Failed to retrieve current value from redis")
    return 0
if current_count is None:
    # Key hasn't been created yet, therefore no hits done so far
    return 0
return int(current_count)
```